File: client-flutter/lib/models/training_type.dart
class TrainingType {
  final String id;
  final String name;
  final int duration; // seconds

  TrainingType({
    required this.id,
    required this.name,
    required this.duration,
  });

  int get durationMinutes => duration ~/ 60;

  factory TrainingType.fromJson(Map<String, dynamic> json) {
    return TrainingType(
      id: json['id']?.toString() ?? '',
      name: json['name']?.toString() ?? '',
      duration: int.tryParse(json['duration']?.toString() ?? '3600') ?? 3600,
    );
  }
}

File: client-ios/SihlClient/Models/TrainingType.swift
import Foundation

/// Pendant zu `models/training_type.dart`.
struct TrainingType: Identifiable {
    let id: String
    let name: String
    let duration: Int      // Sekunden

    var durationMinutes: Int { duration / 60 }

    init(json: [String: Any]) {
        self.id       = "\(json["id"] ?? "")"
        self.name     = (json["name"] as? String) ?? ""
        self.duration = Int("\(json["duration"] ?? 3600)") ?? 3600
    }
}
